apps/cowswap-frontend/src/pages/Claim/InvestmentFlow/hooks/useApproveCallbackFromClaim.ts

```diff
@@ -0,0 +1,59 @@
+import { useMemo } from 'react'
+
+import { V_COW_CONTRACT_ADDRESS } from '@cowprotocol/common-const'
+import { useWalletInfo } from '@cowprotocol/wallet'
+import { TransactionResponse } from '@ethersproject/providers'
+import { Currency, CurrencyAmount, MaxUint256 } from '@uniswap/sdk-core'
+
+import { ClaimType } from 'legacy/state/claim/hooks/types'
+import { EnhancedUserClaimData } from 'legacy/state/claim/types'
+
+import { useApproveCallback } from 'common/hooks/useApproveCallback'
+import { ApprovalState, useApproveState } from 'common/hooks/useApproveState'
+
+type ApproveCallbackState = {
+  approvalState: ApprovalState
+  approve: (summary: string) => Promise<TransactionResponse | undefined>
+  revokeApprove: (summary: string) => Promise<TransactionResponse | undefined>
+}
+
+export function useApproveCallbackFromClaim(claim: EnhancedUserClaimData): ApproveCallbackState {
```

Comment:
This is not a new code, it was extracted from `legacy/hooks/useApproveCallback/index.ts`